File: practica 3/implementacion dart p3/lib/ModeloSOSEMADO/FiltroParteTecnico.dart
import 'Filtro.dart';
import 'Trabajo.dart';
import 'ParteTecnico.dart';

class FiltroParteTecnico extends Filtro{
  void filtrar(Trabajo unTrabajo){
    const double extraTecnico = 0.05;

    ParteTecnico parte = unTrabajo.getParteTecnico();

    var amabilidad = parte.getAmabilidad();
    var profesional = parte.getProfesionalidad();
    var eficacia = parte.getEficacia();


    double mediaParte = ((amabilidad + profesional + eficacia)/3)/10 ;

    double presupuestoAntiguo = unTrabajo.getPresupuesto();
    double nuevoPresupuesto = presupuestoAntiguo * (1+extraTecnico*mediaParte);
    unTrabajo.setPresupuesto(nuevoPresupuesto);
  }
}

File: practica 1/sesion 3/DS_S3_C++Filters/FiltroParteTecnico.cpp
#include <exception>
using namespace std;

#include "FiltroParteTecnico.h"
#include "Trabajo.h"
#include "Filtro.h"

void FiltroParteTecnico::filtrar(Trabajo * unTrabajo) {
	const float extraTecnico = 0.05;			// Porcentaje extra que se da si el tecnico fue bueno

	ParteTecnico * parte = unTrabajo->getParteTecnico();

	int amabilidad	 = parte->getAmabilidad();
	int profesional = parte->getProfesionalidad();
	int eficacia 	 = parte->getEficacia();

	float mediaParte = ((amabilidad + profesional + eficacia)/3)/10;	// Dividir entre 10 para normalizar media (0 -1)

	float presupuestoAntiguo = unTrabajo->getPresupuesto();
	float nuevoPresupuesto = presupuestoAntiguo + (presupuestoAntiguo*extraTecnico) * mediaParte;

	/*Al presupuesto antiguo le sumamos como mucho el 10% de su valor anterior (extraTrabajo)
	  la cantidad exacta a sumar vendra determinada por la media del parte, es decir, si la media
	  es un 10, entonces le sumaremos el 10% al completo
	*/

	unTrabajo->setPresupuesto(nuevoPresupuesto);
}
